core/src/main/java/org/elasticsearch/common/xcontent/XContentParserUtils.java

```diff
@@ -35,6 +36,19 @@ private XContentParserUtils() {
     }
 
     /**
+     * Makes sure that current token is of type {@link XContentParser.Token#FIELD_NAME} and the the field name is equal to the provided one
+     * @throws ParsingException if the token is not of type {@link XContentParser.Token#FIELD_NAME} or is not equal to the given field name
+     */
+    public static void ensureFieldName(XContentParser parser, Token token, String fieldName) throws IOException {
+        ensureExpectedToken(Token.FIELD_NAME, token, parser::getTokenLocation);
+        String current = parser.currentName() != null ? parser.currentName() : "<null>";
```

Comment:
I think if the currentName is null we have a bigger problem. why do we do the comparison even? have you tried passing in fieldName set to <null> ? if the current token is a field name, currentName should not be null ever. I would treat that corner case differently.